src/ray/core_worker/actor_manager.h

```diff
@@ -102,6 +102,14 @@ class ActorManager {
   /// This is used for debugging purpose.
   std::vector<ObjectID> GetActorHandleIDsFromHandles();
 
+  /// Resolve locations of actors that are not persisted to GCS yet.
+  /// https://github.com/ray-project/ray/pull/8679/files
+  /// It is required because core workers persist actor information to
+  /// GCS after it resolves all local dependencies. It means that, there's
+  /// no way for a worker to figure out if the actor is dead or not, which
+  /// can lead a worker to hang forever.
```

Comment:
```suggestion
  /// Periodically check whether the owners are alive for actors whose locations have not yet been persisted to the GCS.
  /// https://github.com/ray-project/ray/pull/8679/files
  /// It is required because the owner persists actor information to the
  /// GCS after it resolves all local dependencies. It means that if the location is not yet in the GCS, we should check whether the owner is still alive to prevent this worker from hanging forever while waiting for the actor's location.
```

Suggested to clarify the edge case and what this method does, but feel free to change the wording!

Reply:
Also, maybe a better name for this would be `MarkPendingActorsFailed`? Since we're not really resolving the actor's location in this method.

Reply:
Yeah I agree. Will change the function name!